templates/addons/calico/values.yaml

```diff
@@ -5,5 +5,5 @@ installation:
     bgp: Disabled
     mtu: 1350
     ipPools:
-    - cidr: 192.168.0.0/16
+    - cidr: 172.16.0.0/12
```

Comment:
this change specifically is a bit concerning... Others might be using this values file at the master branch (eg. `az capi`) and changing this value will break them if they are still using the old cidr value in templates

Reply:
Are these reference templates part of our upgrade contract? If so, yes, this would be non-back-compat.

(But if our reference templates are part of a CAPZ upgrade contract, that's not really sustainable IMO.)

Reply:
@mboersma is `az capi` using the files at main?

Reply:
@CecileRobertMichon no, by default `az capi` uses a local yaml template (which needs to be updated, see Azure/azure-capi-cli-extension#188).

And the CAPZ ref image pipeline uses `az capi` but fetches templates from CAPZ 1.6 currently. I'm updating those to 1.7.3 and then adding MachinePools when time permits.

Reply:
@mboersma that's not what the code is saying: https://github.com/Azure/azure-capi-cli-extension/blob/main/src/capi/azext_capi/custom.py#L653

this is for the helm install of Calico specifically. It would be an issue if the main branch values file changes but the templates are still using the old cidr.

/hold

Reply:
There is now a [v0.1.4 release](https://github.com/Azure/azure-capi-cli-extension/releases/tag/v0.1.4) of `az capi` that [pins those files](https://github.com/Azure/azure-capi-cli-extension/pull/191) to the release-1.7 branch instead. So it shouldn't be blocking this PR any longer.

Reply:
@mboersma do we really want to pin the CNI implementation to a static release of CAPZ indefinitely?